
# A knowledgebase of the roles of genes in different cancers



```{r cancermineSetup, include=FALSE}
source('cancermine/dependencies.R')

knitr::opts_chunk$set(echo = TRUE)
```

```{r cancermineFigures, echo=F}
pdf(file="sideeffects.pdf")
#source('summaryStats.R')
#read_chunk('fig_annotationAndModel.R')
source('cancermine/fig_annotationAndModel.R')
source('cancermine/fig_textminedOverview.R')
#source('fig_inference.R')
source('cancermine/fig_publicationRateAndMore.R')
source('cancermine/fig_time.R')
source('cancermine/fig_accrualAndDualRoles.R')
source('cancermine/fig_comparisons.R')
source('cancermine/fig_clustersAndTCGA.R')
invisible(dev.off())
```

## Introduction

As sequencing technology gets closer to the clinic, more cancer genomic data is used for clinical decision making as part of precision medicine efforts. Interpreting individual cancer samples requires knowledge of which mutations are important in cancer development. The importance of a particular mutation depends on the role of the associated gene and the specific cancer type. The terms driver, tumor suppressor and oncogene are commonly used to respectively denote a gene (or aberrated form of the gene) that is involved in the development of the cancer, reduces the likelihood of cancer developing or promotes the development of cancer [@vogelstein2013cancer]. A deletion or loss-of-function mutation in a tumor suppressor gene associated with the cancer type of the sample is potentially an important event for this cancer. Furthermore, amplifications and gain-of function mutations in oncogenes, and any somatic activity in known driver genes may be valuable information in understanding the mutational landscape of a given cancer sample. This knowledge can then help select therapeutic options and improve our understanding of markers of resistance in the particular cancer type.

A variety of methods exist to identify a gene as a driver, oncogene or tumor suppressor given a large set of genomic data. Many methods use the background mutation rate and gene lengths to calculate a p-value for the observed number of somatic events in a particular gene [@kristensen2014principles]. Other studies use the presence of recurrent somatic deletions or low expression to deduce that a gene is a tumor suppressor[@cheng2017pan]. In-vitro studies that examine the effect of gene knockdowns on the cancer’s development are also used [@zender2008oncogenomics].

Structured databases with information about the role of different genes in cancer, specifically as drivers, oncogenes or tumor suppressors, are necessary for automated analysis of patient cancer genomes. The Cancer Genome Atlas (TCGA) project has provided a wealth of information on the genomic landscape of over 30 types of primary cancers[@weinstein2013cancer]. Data from TCGA (and other resources) are presented in the IntOGen resource to provide easy access to lists of driver genes [@gonzalez2013intogen]. The Cancer Gene Census has been curated using data from COSMIC to provide known oncogenes and tumor suppressors [@futreal2004census] but faces the huge cost of manual curation. Other resources that provide curated information about cancer genes include TSGene [@zhao2015tsgene] and OnGene [@liu2017ongene] but do not match them with specific cancer types. There are also several other resources that are no longer accessible for unknown reasons (NCI Cancer Gene Index, MSKCC Cancer Genes database and Network of Cancer Genes [@an2014ncg]).

Text mining approaches can be used to automatically curate the role of genes in cancer, by identifying mentions of genes and cancer types, and extracting their relations from raw text. Machine learning methods have shown great success in building protein protein interaction (PPI) networks using such data [@szklarczyk2016string]. We present CancerMine (http://bionlp.bcgsc.ca/cancermine/), a robust and regularly updated resource that describes drivers, oncogenes and tumor suppressors in all cancer types using the latest ontologies.

## Methods

### Corpora Processing

The PubMed and PubMed Central Open Access (PMCOA) subsets were downloaded using the PubRunner framework from the NCBI FTP websites. They were then converted to BioC format using PubRunner’s convert functionality. This strips out formatting tags and other metadata and retains the Unicode text of the title, abstract and for PMCOA, the full article. The source of the text (title, abstract, article) is also encoded.

### Entity recognition

Lists of cancer types and gene names were built using a subset of the Disease Ontology (DO) and NCBI gene lists. These were complemented by matching to the Unified Medical Language System (UMLS). For cancer types, this was achieved using the an associated ID in DO or through exact string matching on the DO item title. For gene names, the Entrez ID was used to match with UMLS IDs. The cancer type was then associated with a DO ID, and the gene names were associated with their HUGO gene name. These cancer and gene lists were then pruned with a manual list of stop-words with several custom additions for alternate spellings/acronyms of cancers. All terms with less than four letters were removed.

The corpus text was loaded in BioC format and processed using the Kindred Python package which, as of v2.0, uses the Spacy IO parser. Using the tokenization, entities were identified through exact string matching against tokens. Longer entity names with more tokens were prioritised and removed from the sentence as entities were identified. Fusion terms (e.g. BCR-ABL1) were identified by finding gene names separated by a hyphen or slash. Non-fusions (e.g. HER2/neu) were then identified when two genes with matching HUGO IDs were attached and combined to be a single non-fusion gene entity. Genes mentioned in the context of pathways were also removed (e.g. MTOR pathway) using a list of pathway related keywords.

### Sentence selection

After Kindred parsing, the sentences with tagged entities were searched for those containing at least one cancer type and at least one gene name. These sentences were then filtered using the terms “tumor suppress”, “oncogen” and “driv” to enrich for sentences that were likely discussing these gene roles.

### Annotation

1,600 of the sentences were then randomly selected and output into the BioNLP Shared Task format for ingestion into an online annotation platform. This platform was then used by three expert annotators who are all PhD students actively engaged in precision cancer projects. The platform presents each possible pair of a gene and cancer and the user must annotate this as driving, oncogene and tumor suppressor. The first 100 sentences were used to help the users understand the system, evaluate initial inter-annotator agreement, and adjust the annotation guidelines (available at the Github repository). The results were then discarded and the complete 1,500 sentences were annotated by the first two annotators. The third annotator then annotated the sentences that the first two disagree. The inter-annotator agreement was calculated using the F1-score. A gold corpus was created using the majority vote of the annotations of the three annotators.

### Relation extraction

75% of the 1500 sentences were used as a training set and a Kindred relation classifier was trained with an underlying logistic regression model for all three gene roles (Driver, Oncogene and Tumor_Suppressor). The threshold was varied to generate the precision-recall curves with evaluate on the remaining 25% of sentences. With the selection of the optimal thresholds, a complete model was trained using all 1,500 sentences. This model was then applied to all sentences found in PubMed and PMCOA that fit the sentence requirements. The associated gene and cancer type IDs were extracted, entity names were normalized and the specific sentence was extracted.

### Web portal

The resulting cancer gene roles data were aggregated by the triples (gene, cancer, role) in order to count the number of citations supporting each cancer gene role. This information was then presented through tabular and chart form using a Shiny web app.

### Resource comparisons

The data from the Cancer Gene Census (CGC), IntOGen, TS and ONGene resources were downloaded for comparison. HUGO gene IDs in CancerMine were mapped to Entrez gene IDs. CGC data was mapped to Disease Ontology cancer types. Oncogenes and tumor suppressors were extracted using the presence of "oncogene" or "TSG" in the "Role in Cancer" column. The mapped CGC data was then compared against the set of oncogenes and tumor suppressors in CancerMine. IntOGen cancer types were manually mapped to corresponding CancerMine cancer types and compared against all of CancerMine. The TSGene and ONGene gene sets were compared against the CancerMine gene sets without an associated cancer type.

### CancerMine profiles and TCGA analysis

For each cancer type, the citation counts for each gene role that were in the top `r paper.clusteringTopCancerCount` cancer genes were then log10-transformed and rescaled so that the most important gene had the value of 1 for each cancer type. Gene roles with values lower than 0.2 for all cancer types were trimmed. The top `r paper.clusteringTopCancerCount` cancer types and genes were then hierarchical clustered for the associated heatmap.

The open-access VarScan somatic mutation calls for the seven TCGA projects (BRCA,COAD,LIHC,PRAD,LGG,LUAD,STAD) were downloaded from the GDC Data Portal (https://portal.gdc.cancer.gov). They were filtered for mutations that contained a stop gain or were classified as probably damaging or deleterious by PolyPhen. Tumor suppressor specific CancerMine profiles were generated that used all tumor suppressors for each cancer type. The citation counts were again log10-transformed and the vector for each cancer type was normalized to produce the CancerMine tumor suppressor profile. Each TCGA sample was represented as a binary vector matching the filtered mutations. The dot-product of a sample vector and a CancerMine profile vector produced the sum of citation weightings and gave the score. For each sample, the score was calculated for all seven cancer types and the highest score was used to label the sample. A sample that did not contain tumor suppressor mutations associated with any of the seven profiles was labelled as 'none'.


## Results

### Role of `r paper.geneCount` unique genes elucidated in `r paper.cancerCount` cancer types

```{r annotationAndModel, echo=F, eval=T, fig.width=8, fig.asp=1.0,  fig.cap='(ref:annotationAndModel)'}
grid.arrange(fig_annotationAndModel)
```

(ref:annotationAndModel) The supervised learning approach of CancerMine involves manual annotation by experts of sentences discussing cancer gene roles. Machine learning models are then trained and evaluated using this data set. (a) Manual text annotation of 1,500 randomly selected sentences containing genes and cancer types show a similar number of Oncogene and Tumor Suppressor annotations. (b) The inter-annotator agreement (measured using F1-score) was high between three expert annotators. (c) The precision recall curves show the trade-off of false positives versus false negatives. (d) Plotting the precision-recall data in relation to the threshold applied to the classifier’s decision function provides a way to select a high-precision threshold.

The entire Pubmed and Pubmed Central Open Access subset (PMCOA) corpora were processed to identify sentences that discuss a gene and cancer types within titles, abstracts and where accessible full text articles. By filtering for a customized set of keywords, these sentences were enriched for those likely discussing the genes’ role and 1,500 randomly selected sentences were manually annotated by three expert annotators. Using a custom web interface, the annotators tagged sentences that discussed one of three gene roles (driver, oncogene and tumor suppresser) with a mentioned type of cancer (Fig \@ref(fig:annotationAndModel)A). With high inter-annotator agreement (Fig \@ref(fig:annotationAndModel)B), the data was split into 75%/25% training and test splits. A machine learning model was built for each of the three roles and precision-recall curves were generated (Fig \@ref(fig:annotationAndModel)C) using the test set. A high threshold was selected for each gene role in order to provide high-precision prediction with the accepted trade-off of low recall (Fig \@ref(fig:annotationAndModel)D).

```{r textminedOverview, echo=F, eval=T, fig.width=8, fig.asp=1.5,  fig.cap='(ref:textminedOverview)'}
grid.arrange(fig_textminedOverview)
```

(ref:textminedOverview) Overview of the cancer gene roles extracted from the complete corpora. (a) The counts of the three different gene roles extracted. (b) and (c) show the most frequently extracted genes and cancer types in cancer gene roles. (d) The most frequent journal sources for cancer gene roles. (e) illustrates a large number of cancer gene roles have only a single citation supporting it but that a large number (`r paper.triplesWithMultipleCitations`) have multiple citations.

We apply the models to all sentences selected from Pubmed and PMCOA, identifying `r paper.sentenceCount` sentences from `r paper.pmidCount` unique papers that mention gene roles in cancer. We extract the unique gene/cancer pairs for each role (Fig \@ref(fig:textminedOverview)A) and find that `r paper.geneCount` genes and `r paper.cancerCount` cancer types are covered. These capture the commonly discussed cancer genes and types (Fig \@ref(fig:textminedOverview)B/C) from a large variety of journals (Fig \@ref(fig:textminedOverview)D). We provide a coverage of `r paper.diseaseOntologyCoverage`% (`r paper.cancerCount`/`r paper.diseaseOntologyCancerCount`) of the cancer types described in the Disease Ontology [@schriml2011disease]  having at least one gene association. These results are made accessible through a web portal which can be explored through a gene or cancer-centric view. The resulting data are stored with Zenodo for versioning and download. This storage will provide the results in perpetuity. The results are licensed under the Creative Commons Public Domain (CC0) license to allow this data to be easily integrated with precision cancer workflows.

The trade-off of higher precision with lower recall was made based on the hypothesis that there exists a large amount of redundancy within the published literature. The same idea is often stated multiple times in different papers in slightly different ways. Therefore, for frequently stated ideas, a method with lower recall would likely identify at least one occurrence. This hypothesis is supported by the frequent extraction of commonly-known gene roles such as ERBB2 as an oncogene in breast cancer (`r paper.erbb2_oncogene_breast` citations) and APC as a tumor suppressor in colorectal cancers (`r paper.apc_tumorsuppressor_colorectal` citations). On the other hand, a long tail exists of gene roles with only a single citation – `r paper.triplesWithSingleCitations` of `r paper.triplesWithAnyCitations` (`r paper.percSingleCitations`%) of extracted cancer gene roles (Fig \@ref(fig:textminedOverview)E). For researchers that are accepting of a higher false positive rate, we provide an additional dataset using a lower prediction threshold and estimated average precision and recall of `r paper.lenient_avg_precision` and `r paper.lenient_avg_recall` respectively. The individual prediction score, akin to probabilities, are included so that users can further prune results if needed.

### `r paper.novelTumorSuppressorPerMonth2017` new tumor suppressors are published in literature each month

```{r time, echo=F, eval=T, fig.width=8, fig.asp=1.0, fig.cap='(ref:time)'}
grid.arrange(fig_time)
```

(ref:time) Examination of the sources of the extracted cancer gene roles with publication date. (a) More cancer gene roles are extracted each year but the relative proportion of novel roles remains roughly the same. (b) Roles extracted from older papers tend to focus on oncogenes, but mentions of driver genes have become more frequent since 2010. (c) The full text article is becoming a more important source of text mined data. (d) Different sections of the paper, particularly the Introduction and Discussion parts, are key sources of mentions of cancer gene roles (d).

By examining the publication dates of the articles containing the mined cancer gene roles, we can see that the rate of published cancer gene roles is increasing over time (Fig \@ref(fig:time)A). In 2017, there were `r paper.rolesIn2017` mentions of cancer gene roles in publications, translating to over ~`r paper.avgRolesPerMonth` each month. Approximately `r paper.notnovel2017Perc`% of these are gene roles that have been published previously, but more importantly, the remaining `r paper.novel2017Perc`% are novel. A breakdown by the role shows that oncogenes and tumor suppressors greatly outnumber driver genes. In 2017, `r paper.driver2017`, `r paper.oncogene2017` and `r paper.tumorsuppressor2017` genes were mentioned as drivers, oncogenes and tumor suppressors (Fig \@ref(fig:time)B). Combining this data, we find that there were, on average, `r paper.novelDriverPerMonth2017` novel drivers, `r paper.novelOncogenePerMonth2017` novel oncogenes and `r paper.novelTumorSuppressorPerMonth2017` novel tumor suppressors described in literature each month. This emphasizes the need to keep these text mining results up-to-date at a frequency of less than a year. To this end, we have integrated the CancerMine resource with the PubRunner framework to execute intelligent updates once a month (paper forthcoming - https://github.com/jakelever/pubrunner).  

Unhindered access to the full-text of articles for text mining purposes remains a key challenge. A larger number of cancer gene role mentions are extracted from the full text (`r paper.articleCount`) than from the abstract alone (`r paper.abstractCount`), with a smaller number extracted from the titles (`r paper.titleCount`). As can be seen in Fig \@ref(fig:time)C, the number extracted from full text articles is increasingly dramatically over time. This is likely linked to the increasing number of publications included in the PubMed Central Open Access subset. This strengthens the need for publishers to provide open access and for funding agencies to require publications in platforms that allow text mining. From the full text articles, we extract, where possible, the in-text location of the relationship captured within the paper (Fig \@ref(fig:time)D). Interestingly, a substantial number of the mentions are found in the Introduction section, suggesting that the cancer gene’s role is usually discussed as background information and not a result of the paper. Knowing the subsection that a relationship is captured from can be valuable information for CancerMine users, since a user can then quickly ascertain if the discussed cancer role is prior knowledge or a likely result from the publication. This also highlights the important point that the scientific quality of a paper cannot be verified automatically by text mining technologies, since these methods rely on the statements made by the original author. Hence, any use of text-mined resources will always require users to access the original papers to evaluate the assertion of a gene’s role in a particular cancer.

```{r accrualAndDualRoles, echo=F, eval=T, fig.width=8, fig.asp=1.5, fig.cap='(ref:accrualAndDualRoles)'}
grid.arrange(fig_accrualAndDualRoles)
```

(ref:accrualAndDualRoles) Cancer gene roles discussed first discussed many years ago have a longer time to accrue further mentions (a). Some cancer gene roles grow substantially in discussion while others fade away (b). Cancermine can further validate the dual roles that some genes play as oncogenes and tumor suppressive (c).

Cancer gene roles that are first mentioned at earlier timepoints have more time to accrue additional citations (Fig \@ref(fig:accrualAndDualRoles)A). Thus, it is no surprise that while most cancer gene roles have less than 10 associated citations, those with very large citation counts tend to be published over 10 years ago. For instance, ERBB2’s role as an oncogene in breast cancer is first extracted from a publication in `r paper.erbb2_oncogene_breast_firstYear` and has accumulated `r paper.erbb2_oncogene_breast` citations in literature since then. However, there are some cancer gene roles that were first extracted from publications within the last decade but have already accrued a great number of additional mentions. For instance, KRAS driving non-small cell lung carcinoma is first extracted from a paper published in `r paper.kras_driver_nsclc_firstYear`, and already has `r paper.kras_driver_nsclc` other papers mentioning this role since. Lastly, there are `r paper.forgottenRole_count` cancer gene roles that are mentioned in literature before 2000, but are not extracted in papers after that period. The most frequently mentioned cancer gene role that reflects this pattern is `r paper.forgottenRole_gene` as a `r paper.forgottenRole_role` in `r paper.forgottenRole_cancer`, with `r paper.forgottenRole_before2010` papers mentioning it before 2000 but no further citations afterwards.

With the knowledge of date of publication, we have gleaned a historical perspective on the gene relations captured in literature. For example, the oldest biomarker extracted is MYB as an oncogene in leukemia, first mentioned in an abstract in 1982 [@radke1982transformation]. Fig \@ref(fig:accrualAndDualRoles)B summarizes three trends of citations that we observe, as exemplified by three distinct cancer gene relationships. MYC was cited as an oncogene in plasmacytoma in the mid-eighties and mentioned in multiple citations at the time. However, it is rarely discussed mid-nineties onwards. KLF6 as a tumor suppressor in prostate cancer was first mentioned after the turn of the century but is mentioned infrequently after that. As discussed previously, ERBB2 as an oncogene in breast cancer is the most cited cancer gene role and sees regularly increasing citations year-on-year.

The cancer type is important when trying to understand the context of somatic mutations. This is underscored by gene examples such as NOTCH1. NOTCH1 is a commonly-cited example of a gene that behaves as an oncogene in one cancer (acute T cell leukemia) and as a tumor suppressor in another (head and neck squamous cell carcinoma)[@radtke2003role]. We further validate CancerMine by querying the resource for the set of genes that are (i) strongly identified as a oncogene in at least one cancer type (>`r paper.dualRole_percRequired`% of >=`r paper.dualRole_citationsRequired` citations) and (ii) strongly identified as a tumor suppressor in at least one other cancer type. This method successfully identifies NOTCH1 along with several other genes that play dual roles in different cancer types (Fig \@ref(fig:accrualAndDualRoles)C).

### Text mining provides greater coverage than manually curated resources

```{r comparisons, echo=F, eval=T, fig.width=8, fig.asp=1.2, fig.cap='(ref:comparisons)'}
grid.arrange(fig_comparisons)
```

(ref:comparisons) The Cancermine resource contains substantially more cancer gene associations than other resources and shows reasonable overlap with the Cancer Gene Census resource (a). Surprisingly few of the cancer gene associations are overlapping between the IntOGen resource and CancerMine (b). CancerMine overlaps substantially with the genes listed in the TSGene (c) and ONGene(d) resources.

The Cancer Gene Census (CGC) [@futreal2004census] provides manually curated information about oncogenes and tumor suppressors in `r paper.cgcCancerCount` types of cancer. CGC contains cancer types as raw text, so automated normalization methods were employed to link CGC with the Disease Ontology. CancerMine overlaps with roughly a quarter of the oncogenes and tumor suppressors in the CGC when comparing specific cancer types (Fig \@ref(fig:comparisons)A). Furthermore, it contains information on `r paper.geneCount` genes (compared to `r paper.cgcGeneCount` in CGC) and `r paper.cancerCount` cancer types (compared to `r paper.cgcCancerCount` in CGC). Our resource also provides the sentence in which the gene role is discussed, and citations that link to the corresponding published literature are made available to help the user easily evaluate the evidence supporting the gene’s role.

The IntOGen resource leverages a number of cancer sequencing projects, including the Cancer Genome Atlas (TCGA) to index the inferred genes containing driver mutations. A comparison of the genes with their cancer types in Cancermine shows surprising differences (Fig \@ref(fig:comparisons)B). CancerMine includes a much larger set of genes but has little overlap with the IntOGen resource. This suggests that many of the genes identified through the projects included in IntOGen are not yet frequently discussed in the literature with the specific cancer types in the IntOGen resource.

ONGene and TSGene2 provide lists of oncogenes and tumor suppressors. Unfortunately these gene names are not associated with specific cancer types , an essential aspect for precision oncology. When trying to differentiate between driving and passenger mutations, the lack of cancer type context would likely cause a high false positive rate. CancerMine contains ~`r paper.ongeneOverlapPerc`% of the genes in ONGene and ~`r paper.tsgeneOverlapPerc`% of TSGene2, and contains substantially more genes than both resources (Fig \@ref(fig:comparisons)C/D). These results lend more weight to the use of automated text mining approaches for the population of knowledge bases, since no curation is required to keep the resource up-to-date.

### CancerMine provides insights into cancer similarities and cancer gene function

```{r clustersAndTCGA, echo=F, eval=T, fig.width=10, fig.asp=1.0, fig.cap='(ref:clustersAndTCGA)'}
grid.arrange(fig_profilesAndTCGA)
#plotHeatmap()
#source('cancermine/fig_clusters.R')
```

(ref:clustersAndTCGA) CancerMine data allows the creation of profiles for different cancer types using the number of citations as a weighting for each gene role. (a) The similarities between the top `r paper.clusteringTopCancerCount`  cancer types in CancerMine are shown through hierarchical clustering using weights from the top `r paper.clusteringTopCancerCount` cancer gene roles. (b) All samples in seven TCGA projects are analysed for loss-of-function mutations compared with the CancerMine tumor suppressor profiles and matched with the closest profile. Percentages shown in each cell are the proportion of samples in each project that most closely match the corresponding CancerMine profile. Samples that match no tumor suppressor is these profiles are assigned to none. The TCGA projects are breast cancer (BRCA), colorectal adenocarcinoma (COAD), liver hepatocellular carcinoma (LIHC), prostate adenocarcinoma (PRAD), low grade glioma (LGG), lung adenocarcinoma (LUAD) and stomach adenocarcinoma (STAD).

Oncology often takes an organ-centric view of cancer types which is reflected as demonstrated by the numerous disease ontologies that exist for the categorization and nomenclature of cancer including the Disease Ontology used in this project. However, modern medicine is beginning to look at some cancers based purely on the genetic underpinnings, developing basket trials and approving treatment regimens based on genetic indications only (as shown with the successful approval of Pembrolizumab for PD-1 positive cancer patients). The CancerMine resource allows us to create a gene-centric view of cancers, by clustering cancers based on the role of different genes. A gene-centric view can make it easier to see which treatment regimes could be transferred to other genetically-similar cancer types. To allow for visualisation, we selected the top `r paper.clusteringTopCancerCount` cancers (based on citation count in CancerMine) and extracted the number of citations mentioning the role of top `r paper.clusteringTopCancerCount` genes. This produces a profile for each cancer type showing the importance of each gene and their associated role. A heatmap that illustrates this for the top `r paper.clusteringTopCancerCount` cancer types and genes is shown in Figure \@ref(fig:clustersAndTCGA)a.

The clustering puts biologically similar cancers together, for example colorectal with colon cancer and malignant glioma with glioblastoma multiforme. Some of these clusters also highlight known gene-cancer associations, for example, lung cancer, non-small cell lung carcinoma and lung adenocarinoma all cluster together, and are heavily associated with the KRAS and EGFR oncogenes. Infact, the strong cluster of genes on the left side separates cancers that are strongly associated with KRAS, EGFR, and TP53 (such as lung cancer) from those that are less so (such as thyroid medullary carcinoma). Put together, this approach is able to explain biological similarity of cancer types using shared gene associations. As an example, leukemia clusters closely with the more specific subtype, AML, and it is evident that this is driven by extracted associations of these cancers with MYC, ABL1 and many other genes. Several gene associations are noticeably low frequency compared to overall patterns, for instance KRAS in glioblastoma multiforme (GBM). While there are a small number of papers discussing KRAS in GBM, it is an infrequently discussed gene compared to EGFR and PTEN. Overall this visualisation presents an easy method to explore the similarities and differences between cancer types.

In order to validate the cancer genes identified in CancerMine, we compare results to somatic mutation data from the Cancer Genome Atlas (TCGA) project. We hypothesis that the genes denoted to be tumor suppressors would likely be affected by loss-of-function mutations. Using CancerMine profiles based on tumor suppressor genes, we compare somatic calls for all samples with mutation data within the seven TCGA projects. For each sample, we match the somatic calls against the set of CancerMine tumor profiles and sum the importance of the tumor suppressors found to be mutated. Figure \@ref(fig:clustersAndTCGA)b shows the percentages for each TCGA project that optimally match to each CancerMine profile. Four of the seven TCGA projects (COAD, PRAD, LGG and LUAD) have their highest proportion matches with the corresponding cancer type. Furthermore for each CancerMine profile, the highest percentage corresponds to the corresponding TCGA project for all seven types. Interestingly a third of the prostate cancer samples do not have any LoF mutations that match against any tumor suppressors for any of the seven types, suggesting that prostate cancer tumor suppressors are disabled through other mechanisms or that there are more tumor suppressors involved which have not been captured by CancerMine.

The glioma (LGG) result is the most prominent with `r paper.tcga_LGG_glioma_perc`% of TCGA LGG samples being most closely identified with the CancerMine malignant glioma profile. This is largely due to the high rate of IDH1 (390/503) mutations identified in the LGG cohort. For the breast cancer (BRCA) project, the highest number of matches corresponds to the lung cancer profile with almost twice the number than the breast cancer profile. We find that this is due to the emphasis on TP53 within the lung cancer profile which when removed causes the breast cancer profile to best fit more samples than the lung cancer (173 versus 110 of 976 samples). While this data would not be enough for a tumor type classifier on its own, this results shows there is substantial signal that can be leveraged for interpreting the genomic data and could be combined further with other mutational data. This is underscored when examining breast cancer tumor suppressors with only a single citation, genes that are hypothetically not well known to be tumor suppressors in breast cancer. 10 of these genes (AHNAK, ARID1A, ARID1B, FGFR2, KDM5B, KMT2C, MAP3K1, PRKDC, SPEN & TBX3) are mutated in at least 10 TCGA BRCA samples providing extra strength for the importance of these genes in breast cancer. In fact, the mechanism through which KMT2C inactivation drives tumorgenesis was recently elaborated in ER-positive breast cancer [@gala2018kmt2c].

## Discussion

This work contributes a much needed resource of known drivers, oncogenes and tumor suppressors in a wide variety of cancer types. The text mining approach taken is able to discern complicated descriptions of cancer gene roles with a high level of precision. This provides for a continually updated resource with little need for human intervention. There are several limitations to this approach that present interesting but challenging alleys for further investigation. Firstly, this method focuses on single sentence extraction due to the challenge of anaphora and coreference resolution across sentences. Previous work has shown the high false positive rate that frequently occurs when identifying knowledge across multiple sentences [@lever2016verse]. Our approach requires that authors discuss the gene name, role and cancer name all within the same sentence. This is a problem of writing style and probabilities that gets greatly diluted with the large number of publications processed. Furthermore our approach focuses on individual genes in isolation and is unable to capture complex interactions between cancer genes discussed in papers, e.g. mutual exclusivity. More of these complex relationships will likely be identified in the future and play a part in interpreting the somatic events in an individual cancer patient. Text mining approaches face growing challenges with extracting complex events like these, which may span multiple sentences or even paragraphs.

One important concept when interpreting CancerMine data is that our methodology does not force a definition of a driver, oncogene or tumor suppressor and relies on the assertion of individual authors. A decision was made to not extract discussion of genes frequently mutated in cancer. This was due to the acknowledged problem of huge genes (e.g. TTN) that frequently accrue many somatic mutations but likely don’t play a part in cancer. Instead we rely on the authors’ assertions of the role a gene plays in cancer. The level of evidence differs greatly as some assertions are based on intervential studies (e.g. knockdowns) while others may make assertions based on observational studies (e.g mutation frequency or expression experiments).

As has been noted, many attempts have been made to create a knowledge base of this topic. Hosting the data through Zenodo and the code through Github provides a level of continuity that will guarantee that the project code and data stay accessible for the foreseeable future. Furthermore the PubRunner integration makes it easier to keep the results up-to-date. All data and analysis for this chapter is open source and documented. We hope others will explore this data in order to infer new knowledge of cancer types and their associated genes.
